source/js/iframeAcceptance.js

```diff
@@ -1,68 +1,75 @@
 }
 
-export default () => addEventListener('DOMContentLoaded', () => {
-    if (window.innerHeight > window.innerWidth) {
-        [...document.querySelectorAll('.embed__ratio--16-9')].forEach(embed => {
-            embed.classList.replace('embed__ratio--16-9', 'embed__ratio--1-1');
-        });
+/* Loops through an reveal every URL-host matching local storage */
+const revealContentLoop = () => {
+    [...document.querySelectorAll('.js-suppressed-content')]
+    .forEach(contentWrapper => {
+        if (contentWrapper.classList.contains('js-suppressed-content--none')) {
+            const iframeUrl = new URL(contentWrapper.getAttribute('data-src'));
+            if(acceptedSuppliers.includes(iframeUrl.host)) {
+                revealContent(contentWrapper);
+            }  
+        }
+    }); 
+}
+
+const handleEvents = (contentWrapper) => {
+    /* Sets local storage  */
+    setLocalStorage(contentWrapper);
+
+    /* Modifiers (else equals "no modifier") */
+    if (contentWrapper.classList.contains('js-suppressed-content--video')) {
+        revealContent(contentWrapper);
+    } else {
+        revealContentLoop();
     }
-    if (acceptedSuppliers.length > 0 && 
-        document.querySelectorAll('.js-suppressed-iframe-wrapper').length > 0 ) {
-        [...document.querySelectorAll('.js-suppressed-iframe-wrapper')].forEach(iframeWrapper => {
-            const iframe = iframeWrapper.querySelector('iframe');
-            const iframeUrl = new URL(iframe.getAttribute('data-src'));
-            const iframeWrapperParent = iframeWrapper.parentNode;
-            if (acceptedSuppliers.includes(iframeUrl.host)) {
-                 if (iframeWrapperParent.classList.contains('embed__ratio--1-1')) {
-                    iframeWrapperParent.classList.replace('embed__ratio--1-1', 'embed__ratio--16-9');
-                } 
-                iframe.setAttribute('src', iframeWrapper.querySelector('iframe').getAttribute('data-src'));
-                iframeWrapper.querySelector('.js-suppressed-iframe-prompt').style.display = 'none';
-                iframeWrapper.classList.remove('js-suppressed-iframe-wrapper'); 
+}
+
+/* Adds click events for all suppressed content */
+const setEvents = () => {
+    [...document.querySelectorAll('.js-suppressed-content')]
+    .forEach(contentWrapper => {
+        contentWrapper.querySelector('.js-suppressed-content-description').style.display = "block";
+        const buttonEl = contentWrapper.querySelector('[js-suppressed-content-accept]');
+        buttonEl.addEventListener('click', () => {
+            handleEvents(contentWrapper);
+        });  
+    });
+}
+
+export default () => addEventListener('DOMContentLoaded', () => {
+    if (acceptedSuppliers.length > 0 && document.querySelectorAll('.js-suppressed-content').length > 0 ) {
+        /* No modifiers */
+        [...document.querySelectorAll('.js-suppressed-content--none')].forEach(contentWrapper => {
```

Comment:
What if the class .js-suppressed-content--none dosent exist? There are no undefined check here.

Reply:
That is true, i will fix this one!